qiskit/providers/ibmq/api_v2/session.py

```diff
@@ -0,0 +1,122 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2018, 2019.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Session customized for IBM Q access."""
+
+from requests import Session, RequestException
+from requests.adapters import HTTPAdapter
+from urllib3.util.retry import Retry
+
+from .exceptions import ApiError
+
+
+STATUS_FORCELIST = (
+    500,  # Internal Server Error
+    502,  # Bad Gateway
+    503,  # Service Unavailable
+    504,  # Gateway Timeout
+)
+CLIENT_APPLICATION = 'qiskit-api-py'
+
+
+class RetrySession(Session):
+    """Session with retry and handling of IBM Q parameters."""
+
+    def __init__(self, base_url, access_token=None,
+                 retries=5, backoff_factor=0.5,
+                 verify=True, proxies=None, auth=None):
+        """RetrySession constructor.
+
+        Args:
+            base_url (str): base URL for the session's requests.
+            access_token (str): access token.
+            retries (int): number of retries for the requests.
+            backoff_factor (float): backoff factor between retry attempts.
+            verify (bool): enable SSL verification.
```

Comment:
Can we check this with `verify_origin` to make it more explicit?

Reply:
I'm unsure if I'm reading you correctly - do you mean renaming the parameter? In this case it is for keeping the same names that `requests` uses (also for `proxies`, `auth`, etc) , as it felt clearer (with a small bonus for dealing with the transition of qiskit-related code, as we used `verify` since the `IBMQuantumExperience` package days) - please reopen if I misread :grimacing: